lib/rules/block-opening-brace-empty-line-before/__tests__/index.js

```diff
@@ -0,0 +1,214 @@
+"use strict";
+
+const rule = require("..");
+const { messages, ruleName } = rule;
+
+testRule(rule, {
+  ruleName,
+  config: ["always"],
+
+  accept: [
+    {
+      code: "a\n\n{ color: pink; }"
+    },
+    {
+      code: "a\n\n{\ncolor: pink;\n}"
+    }
+  ],
+
+  reject: [
+    {
+      code: "a { color: pink; }",
+      message: messages.expectedBefore(),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "a {\ncolor: pink;\n}",
+      message: messages.expectedBefore(),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "a\n{ color: pink; }",
+      message: messages.expectedBefore(),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "a\n{\ncolor: pink;\n}",
+      message: messages.expectedBefore(),
+      line: 1,
+      column: 1
+    }
+  ]
+});
+
+testRule(rule, {
+  ruleName,
+  config: ["always-single-line"],
+
+  accept: [
+    {
+      code: "a\n{\ncolor: pink;\n}"
+    },
+    {
+      code: "a\n\n{ color: pink; }"
+    },
+    {
+      code: "a {\ncolor: pink;\n}"
+    },
+    {
+      code: "a\n\n{\ncolor: pink;\n}"
+    }
+  ],
+
+  reject: [
+    {
+      code: "a { color: pink; }",
+      message: messages.expectedBeforeSingleLine(),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "a\n{ color: pink; }",
+      message: messages.expectedBeforeSingleLine(),
+      line: 1,
+      column: 1
+    }
+  ]
+});
+
+testRule(rule, {
+  ruleName,
+  config: ["always-multi-line"],
+
+  accept: [
+    {
+      code: "a\n\n{\ncolor: pink;\n}"
```

Comment:
Please, add the following examples:

```
a\r\n\r\n{\r\ncolor: pink;\r\n}
a\n \n{\ncolor: pink;\n}
a\n \n {\ncolor: pink;\n}
a\r\n \r\n {\r\ncolor: pink;\r\n}
a\n\t\n {\ncolor: pink;\n}
a\n\t \n{\ncolor: pink;\n}
```

It should support Windows linebreaks. Also, it should ignore trailing spaces and tabs when checking for an empty line.